tests/phpunit/integration/Modules/Reader_Revenue_ManagerTest.php

```diff
@@ -231,6 +232,92 @@ function ( $domain ) {
 		$this->assertEquals( $expected_filter, urldecode( $filter ) );
 	}
 
+	public function test_sync_publication_onboarding_state_returns_empty_object() {
+		// Set the Search Console option.
+		$this->options->set( Search_Console_Settings::OPTION, array( 'propertyID' => 'http://test.com' ) );
+
+		FakeHttp::fake_google_http_handler(
+			$this->reader_revenue_manager->get_client(),
+			function ( Request $request ) use ( &$filter ) {
+				$url    = parse_url( $request->getUri() );
+				$filter = $url['query'];
+
+				switch ( $url['path'] ) {
+					case '/v1/publications':
+						return new Response(
+							200,
+							array(),
+							json_encode( $this->get_publications_list_response() )
+						);
+				}
+			}
+		);
+
+		$this->reader_revenue_manager->register();
+
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			$this->authentication->get_oauth_client()->get_required_scopes()
+		);
+
+		$result = $this->reader_revenue_manager->set_data(
+			'sync-publication-onboarding-state',
+			array(
+				'publicationID'              => 'ABCDEFGH',
+				'publicationOnboardingState' => 'PENDING_VERIFICATION',
+			)
+		);
+
+		$this->assertNotWPError( $result );
+		$this->assertEquals( (object) array(), $result );
+	}
+
+	public function test_sync_publication_onboarding_state_returns_new_state() {
```

Comment:
As per my comment above please can you update this test name to be more descriptive of the scenario that's being tested.